datalab/datalab_session/data_operations/data_operation.py

```diff
@@ -84,9 +84,9 @@ def get_operation_progress(self) -> float:
 
     def set_output(self, output):
         output_data = {'output_files': output if isinstance(output, list) else [output]}
-        self.set_status('COMPLETED')
-        self.set_operation_progress(1.0)
         cache.set(f'operation_{self.cache_key}_output', output_data, CACHE_DURATION)
+        self.set_operation_progress(1.0)
+        self.set_status('COMPLETED')
```

Comment:
Rearranged the order of these so that settings status to `COMPLETED` is last. This is to prevent a race condition where the frontend polls until `COMPLETED` is reached but could've read that state before the output data was available in the cache and then it would stop polling for the output data and never get it.